internal/cloud/realm/schema.go

```diff
@@ -1,5 +1,60 @@
 package realm
 
-func (c *client) SchemaModels(groupID, appID string) error {
-	return nil
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/10gen/realm-cli/internal/utils/api"
+)
+
+const (
+	syncClientSchemasPathPattern = appPathPattern + "/sync/client_schemas/%s"
+)
+
+const (
+	DataModelLanguageCSharp     = "C_SHARP"
+	DataModelLanguageJava       = "JAVA"
+	DataModelLanguageJavascript = "JAVA_SCRIPT"
+	DataModelLanguageKotlin     = "KOTLIN"
+	DataModelLanguageObjectiveC = "OBJECTIVE_C"
+	DataModelLanguageSwift      = "SWIFT"
+	DataModelLanguageTypescript = "TYPE_SCRIPT"
+)
+
+type SchemaModel struct {
+	ServiceID string             `json:"service_id"`
+	RuleID    string             `json:"rule_id"`
+	Name      string             `json:"model_name"`
+	Namespace string             `json:"collection_display_name"`
+	Imports   []string           `json:"import_statements"`
+	Code      string             `json:"schema"`
+	Warnings  []SchemaModelAlert `json:"warnings"`
+	Error     SchemaModelAlert   `json:"error"`
+}
+
+type SchemaModelAlert struct {
+	Message string `json:"error"`
+	Code    string `json:"error_code"`
+}
```

Comment:
[q] Understand this is coming from the API so nothing to be done here, but maybe something to keep in mind later. Does it make sense that a `Warning`'s alert is considered an `error` vs `message`?

Reply:
I'm with you 💯  on this ... I also don't love we use "error code" terminology for a warning, but I think these warnings/errors are considered more supplemental than anything.  this is all simply reverse-engineering & mirroring what the UI does with these things